woocommerce-gateway-paylike/assets/js/paylike_checkout.js

```diff
@@ -46,16 +52,22 @@ jQuery( function( $ ) {
 						return false;
 				}
 
-				// Don't affect submission if modal is not needed.
-				if ( ! wc_paylike_form.isPaylikeChosen() ) {
+				// Don't open modal if required fields are not complete
+				if ( $( 'input#legal' ).length === 1 && $( 'input#legal:checked' ).length === 0 ) {
+					return false;
+				}
+
+				if ( $( 'input#data-download' ).length === 1 && $( 'input#data-download:checked' ).length === 0 ) {
 					return false;
 				}
 
-				// Don't open modal if required fields are not complete
 				if ( $( 'input#terms' ).length === 1 && $( 'input#terms:checked' ).length === 0 ) {
 					return false;
 				}
 
+
+				if ( ! wc_paylike_form.validateShipmondo() ) return false;
```

Comment:
What a shame that we have to add specific code for another plugin. Shouldn't Shipmondo hook into some general WooCommerce validation that we can then use?

Reply:
The validation is nonstandard, but we are now using it, and some plugins will use it as well. Since this validation is kind of a hack, there's no documentation, and I didn't know about it until I got the tip from a user. So there's a chance that the plugin developers are not aware of the option. This is a safety mechanism for plugins that did not use the validation. Initially, I removed it, but it caused a lot of support tickets, so it's there for backward compatibility.

Reply:
So, Shipmondo can and should change their plugin to use standard validation?

Reply:
The current version is different, they no longer need validation, the default values are not empty, they are instead: "Nearest point available", so the latest updates do not trigger our validation code. This is only something that we've implemented as a fix for a support ticket, a while ago, and this is working for those versions. To answer your question, their plugin has evolved into better defaults to prevent validation issues. 

Normally with WordPress plugins, legacy code remains put for a long period(years) as WordPress users do not frequently update. We support woocommerce versions that were used since the initial release of our plugin 4 years+ ago. We might drop this code at some point, but I would rather do it as a standalone change, and not as part of a bulk change which might cause a big number of support tickets at the same time.

Reply:
Ah, okay, sounds reasonable.